packages/web3-validator/src/validator.ts

```diff
@@ -16,53 +16,212 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+// 				abi[0].startsWith('tuple') &&
+// 				!Array.isArray(abi[0]) &&
+// 				abi[1] &&
+// 				Array.isArray(abi[1])
+// 			) {
+// 				// eslint-disable-next-line prefer-destructuring
+// 				abiType = abi[0];
+// 				abiName = `${level}/${index}`;
+// 				abiComponents = abi[1] as ReadonlyArray<ShortValidationSchema>;
+// 			} else {
+// 				abiType = 'tuple';
+// 				abiName = `${level}/${index}`;
+// 				abiComponents = abi;
+// 			}
+// 		}
+//
+// 		const { baseType, isArray, arraySizes } = parseBaseType(abiType);
+//
+// 		let childSchema: ZodType;
+// 		let lastSchema: ZodType;
+// 		for (let i = arraySizes.length - 1; i > 0; i -= 1) {
+// 			childSchema = convertToZod({
+// 				type: 'array',
+// 				items: [],
+// 				maxItems: arraySizes[i],
+// 				minItems: arraySizes[i],
+// 			});
+//
+// 			lastSchema = childSchema;
+// 		}
+//
+// 		if (baseType === 'tuple' && !isArray) {
+// 			const nestedTuple = abiToZod(abiComponents, abiName);
+// 			nestedTuple.$id = abiName;
+// 			(lastSchema.items as JsonSchema[]).push(nestedTuple);
+// 		} else if (baseType === 'tuple' && isArray) {
+// 			const arraySize = arraySizes[0];
+// 			const item: JsonSchema = {
+// 				$id: abiName,
+// 				type: 'array',
+// 				items: abiToZod(abiComponents, abiName),
+// 				maxItems: arraySize,
+// 				minItems: arraySize,
+// 			};
+//
+// 			if (arraySize < 0) {
+// 				delete item.maxItems;
+// 				delete item.minItems;
+// 			}
+//
+// 			(lastSchema.items as JsonSchema[]).push(item);
+// 		} else if (isArray) {
+// 			const arraySize = arraySizes[0];
+// 			const item: JsonSchema = {
+// 				type: 'array',
+// 				$id: abiName,
+// 				items: convertEthType(String(baseType)),
+// 				minItems: arraySize,
+// 				maxItems: arraySize,
+// 			};
+//
+// 			if (arraySize < 0) {
+// 				delete item.maxItems;
+// 				delete item.minItems;
+// 			}
+//
+// 			(lastSchema.items as JsonSchema[]).push(item);
+// 		} else if (Array.isArray(lastSchema.items)) {
+// 			// Array of non-tuple items
+// 			lastSchema.items.push({ $id: abiName, ...convertEthType(abiType) });
+// 		} else {
+// 			// Nested object
+// 			((lastSchema.items as JsonSchema).items as JsonSchema[]).push({
+// 				$id: abiName,
+// 				...convertEthType(abiType),
+// 			});
+// 		}
+// 	}
+//
+// 	return schema;
+// };
 
 export class Validator {
 	// eslint-disable-next-line no-use-before-define
 	private static validatorInstance?: Validator;
+
 	// eslint-disable-next-line no-useless-constructor, @typescript-eslint/no-empty-function
-	private constructor() {}
 	public static factory(): Validator {
 		if (!Validator.validatorInstance) {
 			Validator.validatorInstance = new Validator();
 		}
 		return Validator.validatorInstance;
 	}
-	private readonly _schemas: Map<string, Validate> = new Map();
-	public getSchema(key: string) {
-		return this._schemas.get(key);
-	}
-
-	public addSchema(key: string, schema: Schema) {
-		this._schemas.set(key, this.createValidator(schema));
-	}
```

Comment:
(its breaking change) For public functions add in change log,

Reply:
Class Validator is not public to users. It is only for internal Web3Validator usage

Reply:
at two places:
https://github.com/web3/web3.js/blob/ad377d16a7e9a8e82857d6102bb4efd95c543e4f/packages/web3-validator/src/default_validator.ts#L20
https://github.com/web3/web3.js/blob/ad377d16a7e9a8e82857d6102bb4efd95c543e4f/packages/web3-validator/src/index.ts#L19

and

https://github.com/web3/web3.js/blob/ad377d16a7e9a8e82857d6102bb4efd95c543e4f/packages/web3/src/index.ts#L342

This will be breaking change for main web3 package as well.